### PR TITLE
Assisted Installer: Move current jobs to edge variant

### PR DESCRIPTION
**Pull request by lranjbar**
Based off discussions around [AGENT-75](https://issues.redhat.com//browse/AGENT-75) and [AGENT-115](https://issues.redhat.com//browse/AGENT-115) we are moving the current jobs into an `__edge` variant configuration file. I've removed the current test from the base configuration files. These will be testing the images going into the OCP payload.

As far as the OCP payload testing the Assisted Installer jobs in the OCP payload test are the nightly periodics ran from `openshift/release`. These jobs are unaffected by this change.

[Assisted Installer the jobs in the OCP 4.10 payload test](https://github.com/openshift/release/blob/8af01b3c5a869ec1b9f5de2ff79b62aa032c2782/core-services/release-controller/_releases/release-ocp-4.10.json#L83-L90)

/assign @osherdp @eliorerz @andfasano

### CODE REVIEW

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    name: release
+    namespace: openshift
+    tag: golang-1.16
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.16
+images:
+- dockerfile_path: Dockerfile.image-service
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml:20]: are we required to define this image both here and in the main config?

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-periodics.yaml [hunk 2]
```diff
@@ -66,18 +68,10 @@ periodics:
   labels:
     ci-operator.openshift.io/cloud: packet-edge
     ci-operator.openshift.io/cloud-cluster-profile: packet-assisted
+    ci-operator.openshift.io/variant: edge
     ci.openshift.io/generator: prowgen
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
-  name: periodic-ci-openshift-assisted-installer-agent-master-subsystem-test-periodic
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-periodics.yaml:71]: Seems like we're missing those kind of notifications

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml [hunk 3]
```diff
@@ -6,17 +6,18 @@ presubmits:
...
     labels:
       ci-operator.openshift.io/cloud: packet-edge
       ci-operator.openshift.io/cloud-cluster-profile: packet-assisted
+      ci-operator.openshift.io/variant: edge
       ci.openshift.io/generator: prowgen
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
-    name: pull-ci-openshift-assisted-installer-agent-master-e2e-metal-assisted
-    rerun_command: /test e2e-metal-assisted
+    name: pull-ci-openshift-assisted-installer-agent-master-edge-e2e-metal-assisted
+    rerun_command: /test edge-e2e-metal-assisted
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml:20]: Worth giving people heads-up that they need to do ``/test edge-<previous_name>`` when this PR gets in

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
+base_images:
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml:1]: this is rather a general question: why do we move existing configuration to ``edge`` variant?
is it to be able to copy a very specific configuration when branching-out to the ``release-x.y`` branches?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml [hunk 5]
```diff
@@ -3,10 +3,6 @@ base_images:
     name: stream8
     namespace: coreos
     tag: "8"
-  golangci-lint:
-    name: golangci-lint
-    namespace: ci
-    tag: v1.37.1
   openshift_release_golang-1.16:
```
Comment (andfasano) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:6]: This can be bumped up to 1.17

Comment (osherdp) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:6]: You're probably right, but I think we should rather not change too much behavior in this PR :)

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml (hunk 4 above)
Comment (andfasano) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml:1]: The discussion was about using the canonical jobs for shipping the images into the ocp release payload, in a similar manner to the other openshift components that are part it (there is also a technical motivation, as it's not possible to specify multiple promotion targets in the same job). Moving to exisitng jobs to the `edge` variant will preseve the current behavior, as their artifacts will be continued to be promoted in their own namespaces

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml (hunk 5 above)
Comment (andfasano) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:6]: Well, if future it will be a taks for the ci-bot :)

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml (hunk 3 above)
Comment (lranjbar) [reply ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml:20]: It was brought up in the arch meeting last week. I will make a team notification in Slack too.

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml [hunk 6]
```diff
@@ -74,25 +76,26 @@ presubmits:
       - name: result-aggregator
         secret:
           secretName: result-aggregator
-    trigger: (?m)^/test( | .* )e2e-metal-assisted,?($|\s.*)
+    trigger: (?m)^/test( | .* )edge-e2e-metal-assisted,?($|\s.*)
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
[the remark this answers is not in the document]
Comment (lranjbar) [reply ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml:81]: I did a file copy... let me go look for those tweaks again. I wish this setting was in the `/config/` and not the auto generated `/jobs/`

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml (hunk 5 above)
Comment (lranjbar) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:6]: Does the ci-bot update that? That would be awesome 😀

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml [hunk 7]
```diff
@@ -3,10 +3,6 @@ base_images:
     name: stream8
```
[the remark this answers is not in the document]
Comment (lranjbar) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:3]: It will probably be taken out when this is replaced by the `Dockerfile.ocp` but also if we stop using it the ci-bot will remove it for us as well. 😀

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml (hunk 1 above)
Comment (lranjbar) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main__edge.yaml:20]: Note: For the `assisted-image-service` this is more about keeping it consistent. The assisted-image-service wasn't slotted for the 4.11 OCP payload but I am trying to keep this as consistent as possible.

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-periodics.yaml (hunk 2 above)
Comment (lranjbar) [reply ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-periodics.yaml:71]: I went through and put the slack notifications back in

File: ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (lranjbar) [reply ci-operator/jobs/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master-presubmits.yaml]: I went through and changed these back to what they were

**Comment by lranjbar** [comment]
Update we did get the following approved names for the OCP payload:

- agent-installer-api-server (assisted-service)
- agent-installer-node-agent (assisted-installer-agent)
- agent-installer-csr-approver  (assisted-installer-controller)
- agent-installer-orchestrator  (assisted-installer)
- agent-installer-image-provider-api-server (assisted-image-service)

I will be making a follow up PR to add in the promotions to the ocp namespace because there are enough changes in this PR already.

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml (hunk 5 above)
Comment (andfasano) [reply ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:6]: Yeah, there's a registry replacer for that, ie https://github.com/openshift/release/commit/48ba4f8d9745cb273048d6c89245e70bd29e1ef3

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master.yaml [hunk 8]
```diff
@@ -1,28 +1,8 @@
...
-    namespace: edge-infrastructure
-    tag: assisted-test-infra
   coreos-centos-stream8:
     name: stream8
     namespace: coreos
     tag: "8"
-  dev-scripts:
-    name: test
-    namespace: ocp-kni
-    tag: dev-scripts
```
Comment (andfasano) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master.yaml:25]: I think we're going to need dev-scripts image, while very likely centos-stream8 not

File: ci-operator/jobs/openshift/assisted-image-service/openshift-assisted-image-service-main-periodics.yaml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 periodics:
 - agent: kubernetes
   cluster: build02
-  cron: 47 3 * * *
+  cron: 15 1 * * *
```
Comment (andfasano) [inline ci-operator/jobs/openshift/assisted-image-service/openshift-assisted-image-service-main-periodics.yaml:4]: out of curiosity, is this periodic change done automatically by the `make jobs`?

Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-image-service/openshift-assisted-image-service-main-periodics.yaml:4]: Yes, I think it generates the random time based on this code: https://github.com/openshift/ci-tools/blob/0086dc59e3c3480a00c9aba76f4ba5f44aec992a/pkg/prowgen/prowgen.go#L186

**Review by osherdp — APPROVED** [review]

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master.yaml (hunk 8 above)
Comment (lranjbar) [reply ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-master.yaml]: This is true. I put it back in but it is likely the CI bots will clean it up for us because there is a job that removes unused images from the inputs stanza. 😄

File: ci-operator/jobs/openshift/assisted-image-service/openshift-assisted-image-service-main-periodics.yaml (hunk 9 above)
Comment (lranjbar) [reply ci-operator/jobs/openshift/assisted-image-service/openshift-assisted-image-service-main-periodics.yaml:4]: Yup it is a generated change

**Comment by lranjbar** [comment]
Rebase complete... there were so many changes I recreated the branch off master. 😅

**Review by osherdp — APPROVED** [review]
